script.js

```diff
@@ -1,4 +1,261 @@
+var myScore = 0;
+var computersScore = 0;
+var numberOfDraws = 0;
+var totalNumberOfRounds = myScore + computersScore + numberOfDraws;
+var myWinningPercentage = (myScore / totalNumberOfRounds) * 100;
+var computersWinningPercentage = 100 - myWinningPercentage;
+
+var currentGameMode = `waiting for user name`;
+var userName = ``;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = ``;
+  if (currentGameMode == `waiting for user name`) {
+    userName = input;
+    currentGameMode = "to select mode";
+    myOutputValue = `Hello ${userName}. Please select your Scissors Paper Stone game mode. Enter either "original" or "korean".`;
+    return myOutputValue;
+  } else if (currentGameMode == `to select mode`) {
+    if (input == `original`) {
+      currentGameMode = `original`;
+    }
+
+    if (input == `korean`) {
+      currentGameMode = `korean`;
+    }
+
+    myOutputValue = `Hi ${userName}, you've selected ${input} mode. Please enter "scissors" or "paper" or "stone" to begin.`;
+    return myOutputValue;
+  } else if (currentGameMode == `original`) {
+    var randomComputershand = computershand();
+
+    // All win cases:
+    if (
+      (input == "scissors" && randomComputershand == "paper") ||
+      (input == "reversed stone" && randomComputershand == "paper")
+    ) {
+      myScore = myScore + 1;
+      computersScore = computersScore + 0;
+      numberOfDraws = numberOfDraws + 0;
```

Comment:
I am not very sure about the need to add 0 to the scores and draw here since there should be no effect on the computer score or the number of draws if the user won the game